packages/api/src/controllers/eventController.ts

```diff
@@ -0,0 +1,86 @@
+import {NextFunction, Request, Response} from 'express';
+import Event from '../models/event';
+
+export const getAllEvents = async (req: Request, res: Response, next: NextFunction) =>
+{
+  try {
+    const events = await Event.find();
+
+    return res.status(200).json({
+      "events": events,
+    });
+  }
+  catch (error)
+  {
+    next(error);
+  }
+}
+
+export const createEvent = async(req: Request, res: Response, next: NextFunction) =>
+{
+  try {
+    const eventData = req.body;
+    const event = await Event.create(eventData);
+
+    return res.status(200).json({
+      "event": event,
+    })
+  } catch (error)
+  {
+    next(error);
+  }
+}
+
+
+export const getEventById = async(req: Request, res: Response, next: NextFunction) =>
+{
+  try {
+    const eventId = req.params.id;
+    const event = await Event.findOne({_id: eventId});
+
+    return res.status(200).json({
+      "event": event,
+    })
+  } catch (error) {
+    next(error)
```

Comment:
Errors get passed to the `next` function but who is doing the error handling? We should be returning `500`s or `400`s range responses based on the situation. AKA, if they didn't specify some required parameter, then that's a `400 Bad Request`, but if we had trouble parsing the JSON on backend side, then we should be returning a `500 Internal Server Error`.

Reply:
Hopefully fixed now